### PR TITLE
Translate poses of nested models inside other nested models (sdf6)

### PR DESCRIPTION
**Pull request by j-rivero**
When `addNestedModel` function is called, it processes links and joints to translate the pose accordingly to parent pose. This was not done for nested models inside the SDF being processed by the function, hence they were not aware of parent pose.

This PR includes nested models in the same way that is doing for links and add a test that fails without the change.
The change was proposed by @azeey. If we are not wrong, the change is not present in newer versions of sdformat after the implementation of frame semantics.

### CODE REVIEW

**Review by scpeters — APPROVED** [review]
this looks good to me

I tried running this test on the `sdf9` branch, and it fails since it looks only at the `<pose>` values without considering the `//pose/@relative_to` attribute. I believe `sdf9` is doing the right thing though, and that the test would pass if the pose values are calculated with the `SemanticPose` API

I believe an equivalent fix was merged to `sdf10` in  #399 and then back-ported to `sdf9` in https://github.com/osrf/sdformat/pull/424

File: test/integration/model/box_with_submodel/model.config [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<model>
+  <name>BoxWithSubmodel</name>
+  <version>1.0</version>
+  <sdf version="1.5">model.sdf</sdf>
```
Comment (azeey) [inline test/integration/model/box_with_submodel/model.config:5]: nit: Should match the version in `model.sdf`
```suggestion
  <sdf version="1.6">model.sdf</sdf>
```

**Review by azeey — APPROVED** [review]
Looks good to me too! We should merge this along with #597 so it doesn't break models with joints that reference links in nested models.

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-25-fortress-edifice-citadel/1343/1

**Pull request merged**